### PR TITLE
feat: implement --host and --port flags as first-class options

### LINKED ISSUE #355: Implement --host and --port options as endpoint aliases
## Overview

Implement `--host` and `--port` flags as first-class options (not aliases) that are mutually exclusive with `--endpoint`, compatible with Google Cloud Spanner CLI.

## Specification

- `--host=<HOST>`: Host on which Spanner server is located
- `--port=<PORT>`: Port number for Spanner connection  
- These flags are mutually exclusive with `--endpoint`
- When only `--port` is specified: connect to `localhost:<PORT>` (for emulator)
- When only `--host` is specified: connect to `<HOST>:443` (standard Spanner port)
- When both are specified: connect to `<HOST>:<PORT>`

## System Variables

- `CLI_ENDPOINT`: Shows the combined endpoint (host:port)
- `CLI_HOST`: Shows the host component
- `CLI_PORT`: Shows the port component

All three system variables are always visible regardless of which flags were used.

## Implementation notes

- Store only Host and Port fields in systemVariables struct
- Remove the Endpoint field to reduce struct fields
- CLI_ENDPOINT becomes a computed property from Host and Port
- Parse `--endpoint` flag into host and port components when provided

## Priority

Low priority - `--endpoint` already provides this functionality

## References

- Parent issue: #342
- [Google Cloud Spanner CLI documentation](https://cloud.google.com/spanner/docs/spanner-cli)
EOF < /dev/null

### PR DESCRIPTION
**Pull request by apstndb**
## Summary
Implements `--host` and `--port` flags as first-class options (not aliases) that are mutually exclusive with `--endpoint`, compatible with Google Cloud Spanner CLI.

## Changes
- Add `--host` and `--port` command-line flags
- Remove `Endpoint` field from `systemVariables` struct and use only `Host` and `Port` fields
- Make `CLI_ENDPOINT` a computed property from `Host` and `Port`
- Add `CLI_HOST` and `CLI_PORT` read-only system variables
- Implement smart endpoint construction logic:
  - `--port` only → `localhost:$PORT` (emulator use case)
  - `--host` only → `$HOST:443` (standard Spanner port)
  - Both specified → `$HOST:$PORT`
- Update all references throughout the codebase

### 1. Endpoint Parsing for IPv6
The implementation includes proper IPv6 address handling in `parseEndpoint()`:
```go
// Handle IPv6 addresses in brackets
if strings.HasPrefix(endpoint, "[") {
    if idx := strings.LastIndex(endpoint, "]:"); idx != -1 {
        host = endpoint[:idx+1]
        portStr := endpoint[idx+2:]
        // ...
    }
}
```

### 2. System Variable Design Pattern
The system variable implementation follows a specific pattern where:
- Read-only variables only have getters
- Read-write variables have both getters and setters
- `CLI_ENDPOINT` is special - it has both getter/setter but operates on the underlying `Host` and `Port` fields

### 3. Test Pattern Discovery
Found that the test framework always calls `Set()` even for getter-only tests. Modified the test to skip `Set()` when `value` is empty:
```go
// Only call Set if value is provided or if testing unimplemented setter
if test.value != "" || test.unimplementedSet {
    err := sysVars.Set(test.name, test.value)
    // ...
}
```

### 4. Session Creation Pattern
All endpoint configuration now flows through `Host` and `Port` fields, with dynamic construction at the point of use:
```go
if sysVars.Host != "" && sysVars.Port != 0 {
    endpoint := fmt.Sprintf("%s:%d", sysVars.Host, sysVars.Port)
    opts = append(opts, option.WithEndpoint(endpoint))
}
```

## Testing
- Added comprehensive tests for flag validation and mutual exclusivity
- Added tests for endpoint parsing including IPv6 support
- Updated all existing tests to work with the new structure
- All tests pass with `go test -short ./...`
- Lint clean with `golangci-lint run`

Fixes #355
EOF < /dev/null

### CODE REVIEW

File: system_variables.go [hunk 1]
```diff
@@ -150,6 +151,37 @@ type systemVariables struct {
...
+	if idx := strings.LastIndex(endpoint, ":"); idx != -1 {
+		host = endpoint[:idx]
+		portStr := endpoint[idx+1:]
+		if p, err := strconv.Atoi(portStr); err == nil {
+			port = p
+		}
+	}
+	
+	return
+}
```
[the remark this answers is not in the document]
Comment (apstndb) [reply system_variables.go]: Fixed in aee00b0. Thank you for catching this critical issue\!

I've replaced the custom parseEndpoint implementation with net.SplitHostPort which properly handles all edge cases including bare IPv6 addresses. I've also added test cases to cover the bare IPv6 scenario.

The implementation now:
- Uses net.SplitHostPort for parsing (fails on bare IPv6 as expected)
- Uses net.JoinHostPort for reconstruction (properly brackets IPv6 addresses)
- Includes test coverage for the edge cases
EOF < /dev/null

Fixed in commit aee00b0.

File: system_variables.go [hunk 2]
```diff
@@ -150,6 +152,29 @@ type systemVariables struct {
...
+		return "", 0
+	}
+
+	p, err := strconv.Atoi(pStr)
+	if err != nil {
+		// The port is not a valid number. The caller should handle this.
+		return h, 0
+	}
+	return h, p
+}
```
[the remark this answers is not in the document]
Comment (apstndb) [reply system_variables.go]: Fixed by making parseEndpoint return errors for invalid endpoints

Fixed in commit 0084064.

File: system_variables.go [hunk 3]
```diff
@@ -717,7 +742,38 @@ var systemVariableDefMap = map[string]systemVariableDef{
...
+			},
+			Setter: func(this *systemVariables, name, value string) error {
+				// Parse endpoint and update host and port
+				host, port := parseEndpoint(unquoteString(value))
+				if host == "" || port == 0 {
+					return fmt.Errorf("invalid endpoint format: %s", value)
+				}
+				this.Host = host
+				this.Port = port
+				return nil
```
[the remark this answers is not in the document]
Comment (apstndb) [reply system_variables.go]: Fixed by updating the CLI_ENDPOINT setter to handle errors from parseEndpoint

Fixed in commit 0084064.
